### DISCUSSION TITLE
How to update extension README?

### DISCUSSION THREAD

**Discussion by damms005** [microsoft/vscode-discussions#1054] [Extension Development QnA] [answered]
I wrongly thought that updating my README file in my GitHub repo and push the changes will cause my [extensions page content](https://marketplace.visualstudio.com/items?itemName=damms005.devdb) to be updated automatically, more so that in my `package.json` file, I defined `"homepage": "https://github.com/damms005/devdb-vscode/blob/main/README.md"`

But clearly this is not so because e.g. the screenshot I used in the **Context Menu entry** section in my [extensions page content](https://marketplace.visualstudio.com/items?itemName=damms005.devdb) is different from what I [currently have in my repo](https://github.com/damms005/devdb-vscode/?tab=readme-ov-file#context-menu-entry)

How can I update the outdated info on my extensions page?

**Comment by tamuratak** [comment]
To update the extension README, you have to upload a new version of your extension. The README file in the extension (the .vsix file) is used for the extension page.

**Reply by damms005** [reply]
Is this not what `vsce publish` does?

**Comment by alefragnani** [comment]
Hi @damms005 ,

In order to update both Marketplace and the Extension's viewlet inside VS Code, you must publish a new version of your extension, using `vsce`. And, just to be clear, the `vsce` tool will use the resources you have _locally_ in the machine where you are running it. This is necessary not only for the markdown content in `README.md`, but also the images/videos that you include on it (located on your extension's repo).

I remember that sometimes we had _excessive delays_ on this publishing process, but in the end, it worked. And I also remember a couple of situations when the content didn't update, even after the publishing process to be sucessfull. In this specific cases, the author had to reach the Markeplace team in order to identify and fix the reason behind the _locked publishing_.

Hope this helps.

**Reply by damms005** [reply]
This is so helpful, @alefragnani 

> the vsce tool will use the resources you have _locally_ in the machine where you are running it

Exactly my thoughts, too. You'll see what I have in [the marketplace](https://marketplace.visualstudio.com/items?itemName=damms005.devdb) is different from what I have in [the repo](https://github.com/damms005/devdb-vscode/edit/main/README.md) - which is what I have locally.

I think the only option left now is to reach the Markeplace team like you suggested. Can you guide me on how I can reach them please?

**Reply by alefragnani** [reply]
Glad to help.

I'm not sure what's the difference you are noticing, as it seems the same content to me, except those `[!NOTE]` render on GitHub. Doing a quick look in the contents, at least.

ps: To reach the VS Marketplace team, I would start looking at their feedback repo https://github.com/microsoft/vsmarketplace.

**Reply by damms005** [reply, marked as answer]
I was being a durk.

Your comment that "it seems the same content" made me to check again and noticed that I did not push the changes to the `main` branch.

Like you said, I also thought that:

> the `vsce` tool will use the resources you have _locally_ in the machine where you are running it

So I never bothered checking the current working git branch.

I just rebased main off the working branch and pushed changes to GitHub few minutes ago. Now marketplace displays the latest changes almost instantly.

Apparently, marketplace displays latest readme found on the main branch irrespective of if a new `vsce` publish was made or not (because I simply just did `git push`, no `vsce` publish anything. zit. nada)

You've been most helpful. Thanks again!

PS: fwiw, the update was just change in screenshot: from [this](https://raw.githubusercontent.com/damms005/devdb-vscode/1c8205dc8199167e6f50b3822b8058ed32fba266/resources/screenshots/context-menu-contribution.png) to [that](https://raw.githubusercontent.com/damms005/devdb-vscode/main/resources/screenshots/context-menu-contribution.png)

**Edit** At least the somewhere in the extensions docs should mention this. Perhaps the [publishing section](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#publishing-extensions)

**Reply by alefragnani** [reply]
Hi @damms005 , 

About the documentation update, since the VS Code Documentation is also open source, I would suggest you to open an issue (or even a PR) in its repo (https://github.com/microsoft/vscode-docs).

Hope this helps

**Reply by damms005** [reply]
I learned shortly after my comment above that the documentation does, indeed, cover this behavior already.

https://code.visualstudio.com/api/working-with-extensions/publishing-extension#marketplace-integration

Thanks for all the help, @alefragnani !
